src/sql/src/pure.rs

```diff
@@ -1648,56 +1745,69 @@ async fn purify_create_table_from_source(
                 PurifiedExportDetails::LoadGenerator { table, output } => (table, output),
                 _ => unreachable!("purified export details must be load generator"),
             };
-
-            let (gen_columns, gen_constraints) = scx.relation_desc_into_table_defs(&desc)?;
+            // We only determine the table description for multi-output load generator sources here,
+            // whereas single-output load generators will have their relation description
+            // determined during statment planning as envelope and format options may affect their
+            // schema.
+            if let Some(desc) = desc {
+                let (gen_columns, gen_constraints) = scx.relation_desc_into_table_defs(&desc)?;
+                *columns = gen_columns;
+                *constraints = gen_constraints;
+            }
```

Comment:
this is a consequence of how load-generators work, where the `generator.views()` method only returns a list of views for multi-output load generators, whereas single-output load-generators are assumed to determine their relations using the encoding/envelope options similar to kafka sources